### ISSUE TITLE
Explainer without Transformer

### ISSUE THREAD

**Issue by soochem** [kubeflow/kfserving#845]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Hello, I've tried two AnchorImage test cases and met errors:
Case 1. Using customized KFServing alibiexplainer image (using alibiexplanier.Dockerfile), 
Alibi explainer dump (alibi v0.3.2), 
and Predictor (model : tensorflow 0.15.2),
I got error when requesting inference. (upstream time out, 504)
Inferenceservice doesn't answer to my curl requests (max try 3, waiting for > 5min).

Case 2. Using same explainer & predictor with **transformer**,
I got 200 response with expected explanation. (in 2 min)

So here's my question: Is it mandatory to use explainer with transformer?
Or they just need more time to inference? (It doesn't make sense, because case 2 goes through more processes and may take longer)
ImageNet example only requires explainer & predictor and I think it should work under the specific conditions.

I also found some error from Alibi version incompatibility.
KFServing alibi explainer from ImageNet example installs alibi 0.3.3dev, which is incompatible with alibi<=0.3.2 (Seldon-model's explainer).

Thank you!

**What did you expect to happen:**
Requesting endpoint:explain --> Getting response 200 --> Results in prediction & anchor

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**
- Istio Version: 1.4.6
- Knative Version: v0.14.0
- KFServing Version: v0.3.0
- Kubeflow version: x
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version: x
- Kubernetes version: (use `kubectl version`): 1.15.9
- OS (e.g. from `/etc/os-release`): macOS, CentOS7

**Comment by cliveseldon** [comment]
Just merged is an update to Alibi 0.4.0: https://github.com/kubeflow/kfserving/pull/803
Can you try with that? @yuzisun are we sure the new image is available on master?

**Comment by soochem** [comment]
Thanks. I used Alibi 0.4.0 & KFServing alibiexplainer v0.3.0
and got :

Traceback (most recent call last):
│       File "/usr/local/lib/python3.7/site-packages/tornado/web.py", line 1701, in _execute
│         result = method(*self.path_args, **self.path_kwargs)
│       File "/kfserving/kfserving/handlers/http.py", line 61, in post
│         response = model.explain(request)
│       File "/alibiexplainer/alibiexplainer/explainer.py", line 74, in explain
│         explanation = self.wrapper.explain(request["instances"])
│       File "/alibiexplainer/alibiexplainer/anchor_images.py", line 45, in explain
│         anchor_exp = self.anchors_image.explain(arr[0], **self.kwargs)
│       File "/usr/local/lib/python3.7/site-packages/alibi/explainers/anchor_image.py", line 365
│         self.instance_label = self.predictor(image[np.newaxis, ...])[0]
│       File "/usr/local/lib/python3.7/site-packages/alibi/utils/wrappers.py", line 33, in __cal
│         pred = np.atleast_2d(self.predictor(x))
│       File "<stdin>", line 1, in <lambda>
│     NameError: name 'model' is not defined
│ [E 200527 01:34:38 web:2250] 500 POST /v1/models/imagenet:explain (127.0.0.1) 551.73ms

I set my predict_fn to None and wonder why I got this. (I'm confused that this "model" means explainer model or model in lambda function)
Thank you!

**Comment by cliveseldon** [comment]
To use Alibi 0.4.0 you would need to use the master version of the alibiexplainer wrapper.
Until the next release you can use https://github.com/kubeflow/kfserving/blob/74685535e20c6d328cce836f0319a33f1350b381/Makefile#L71 to build and push a dev version.

**Comment by yuzisun** [comment]
@soochem we have fixed alibi image in 0.3 release, can you try now?

**Comment by soochem** [comment]
@yuzisun, thank you for replying!
Following your direction, I tried alibi v0.4.0 and it worked fine in imagenet sample case!
Also, my tf.keras model's explainer worked normally with Predictor & Transformer
but showed internal error message in case for InferenceService without Transformer
I guess the output of naive TFServing predictor has different format from that of transformer
I'll figure it out. thanks a lot.

1. 502 Bad Gateway
```
< HTTP/1.1 502 Bad Gateway
< content-length: 4
metadata:
< content-type: text/plain; charset=utf-8
< date: Fri, 04 Sep 2020 07:58:36 GMT
< x-content-type-options: nosniff
< x-envoy-upstream-service-time: 80123
< server: istio-envoy
<
EOF
```
2. 504 upstream request timeout
```
< HTTP/1.1 504 Gateway Timeout
< content-length: 24
< content-type: text/plain
< date: Fri, 04 Sep 2020 08:32:15 GMT
< server: istio-envoy
< x-envoy-upstream-service-time: 600018
```
